### PR TITLE
Changing Groups to Map, fixing #1587 ,I accept the terms of me contributing to this project

### PR DESCRIPTION
**Pull request by aditya-32**
fixing #1587, I accept the terms of me contributing to this project
These changes include changing Group from set to Map and changing its corresponding client and unit tests.

### CODE REVIEW

**Comment by aditya-32** [comment]
@vladak I have created this PR for changing  Groups to Map from Set can you please review this PR
also

File: opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Configuration.java [hunk 1]
```diff
@@ -1144,7 +1155,7 @@ public LuceneLockName getLuceneLocking() {
 
     /**
      * @param value off|on|simple|native where "on" is an alias for "simple".
-     * Any other value is a fallback alias for "off" (with a logged warning).
+     *              Any other value is a fallback alias for "off" (with a logged warning).
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Configuration.java:1158]: remvove that extra whitespace

**Comment by vladak** [comment]
In general this looks sensible. Do you need to include all these extra blank lines in the javadoc ? These create unnecessary noise when reviewing the code.

Have you started the OCA process yet ?

**Comment by aditya-32** [comment]
@vladak Yes have submitted the OCA today

File: opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Configuration.java (hunk 1 above)
Comment (aditya-32) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Configuration.java:1158]: sure will remove those

**Comment by aditya-32** [comment]
@vladak  can you please review this, OCA has been approved!

**Comment by vladak** [comment]
> @vladak can you please review this, OCA has been approved!

Could you change your e-mail address in the author field in the commits so that it matches the one in the OCA ?

**Comment by aditya-32** [comment]
@vladak I have rebased the commits now can you check once if the email is updated 
in git I have the same email I used for signing OCA still the check is failing now

**Comment by vladak** [comment]
> @vladak I have rebased the commits now can you check once if the email is updated in git I have the same email I used for signing OCA still the check is failing now

The author e-mail in the commit is still the one ending with `.local`. You have to amend the commit and force push, I think.

**Comment by aditya-32** [comment]
@vladak  done OCA verified now, thanks for the  help !

**Comment by vladak** [comment]
Thanks !

**Pull request merged**